### PR TITLE
#151: Implement recursion in the target selection strategy and fix notion of "rejected"

### LINKED ISSUE #150: Redefine "reject" and its corresponding count
That counter should probably be incremented only when **no** suitable target was found for an object.

### LINKED ISSUE #151: Study and implement recursion in the target selection strategy
This is to allow for more than one object being transferred at once (up to a pre-determined maximum of recursion depth)

Ideas:
* try to order by "most heavily connected to" relationship;
* try to recurse until no more connected objects are available to the current "cluster";
* try to add some probabilistic trigger (or to the contrary a probabilistic limiter).

### PR DESCRIPTION
**Pull request by ppebay**
Resolve #150 #151

### CODE REVIEW

**Comment by ppebay** [comment]
Continuing discussion of https://github.com/DARMA-tasking/LB-analysis-framework/issues/151#issuecomment-1024621584 here:

**Comment by ppebay** [comment]
Continuing discussion of https://github.com/DARMA-tasking/LB-analysis-framework/issues/151#issuecomment-1024621584 here:

**Comment by ppebay** [comment]
Now confirmed to work with arbitrary-depth recursion (limited by number of objects on rank):

Screen Shot 2022-01-31 at 2 55 05 AM

Converges even faster now as many objects can be transferred at once.

**Comment by ppebay** [comment]
@marcinwrobel1986 this is conflicting all over the place with the recent merges into `develop`. Can you please create order out of this chaos?
Thanks

**Comment by ppebay** [comment]
For good measure, posting here the case for alpha=beta=1 and gamma=0:

Screen Shot 2022-02-03 at 9 14 07 AM

Note: theoretical optimum is 4 for maximum work (4.5 achieved by algorithm).

File: src/Execution/lbsRuntime.py [hunk 1]
```diff
@@ -216,6 +217,27 @@ def information_stage(self, n_rounds, f):
...
+
+        # Pick one object and move it from one list to the other
+        o = random.choice(pick_list)
+        pick_list.remove(o)
+        object_list.append(o)
+
+        # Decide whether criterion allows transfer
+        if c_fct(object_list) < 0.:
+            # Transfer is not possible, recurse further
+            self.recursive_extended_search(pick_list, object_list, c_fct)
```
Comment (lifflander) [inline src/Execution/lbsRuntime.py:236]: Can we make this an configuration option so we can explore both ways easily?

**Review by lifflander — APPROVED** [review]
Looks good to me!

File: src/Execution/lbsRuntime.py (hunk 1 above)
Comment (ppebay) [reply src/Execution/lbsRuntime.py:236]: Yes, creating an issue for that. Thanks.

**Comment by ppebay** [comment]
Tested and verified post-conflict resolution, thanks @marcinwrobel1986

**Pull request merged**